CourseGrab/ViewControllers/Settings/SettingsViewController.swift

```diff
@@ -178,33 +192,123 @@ extension SettingsViewController {
 
 }
 
-// MARK: - Toggle notifications enabled
+// MARK: - Notifications
 
 extension SettingsViewController {
     
+    @objc private func setMobileSwitchOn() {
+        mobileSwitch.isUserInteractionEnabled = false
+        mobileSwitch.isOn = UserDefaults.standard.areNotificationsEnabled
+        UNUserNotificationCenter.current().getNotificationSettings { settings in
+            DispatchQueue.main.async {
+                self.mobileSwitch.isOn = settings.authorizationStatus == .authorized && UserDefaults.standard.areNotificationsEnabled
+                self.mobileSwitch.isUserInteractionEnabled = true
+            }
+        }
+    }
+    
     private func toggleNotificationsEnabled(_ sender: UISwitch) {
-        sender.isUserInteractionEnabled = false
         AppDevAnalytics.shared.logFirebase(MobileAlertPressPayload())
+        sender.isUserInteractionEnabled = false
+        let enable = sender.isOn
         
-        NetworkManager.shared.enableNotifications(enabled: sender.isOn).observe { result in
-            DispatchQueue.main.async {
-                sender.isUserInteractionEnabled = true
-                switch result {
-                case .value(let response):
-                    if response.success {
-                        UserDefaults.standard.areNotificationsEnabled = sender.isOn
-                    } else {
-                        sender.isOn = !sender.isOn
+        UNUserNotificationCenter.current().getNotificationSettings { settings in
+            switch settings.authorizationStatus {
+            case .authorized:
+                NetworkManager.shared.enableNotifications(enabled: enable).observe { result in
+                    DispatchQueue.main.async {
+                        sender.isUserInteractionEnabled = true
+                        switch result {
+                        case .value(let response):
+                            if response.success {
+                                UserDefaults.standard.areNotificationsEnabled = sender.isOn
+                            } else {
+                                sender.isOn = !sender.isOn
+                            }
+                        case .error(let error):
+                            print(error)
+                        }
                     }
-                case .error(let error):
-                    print(error)
+                }
+            case .notDetermined:
```

Comment:
Should we remove this case since there is no difference between `.notDetermined` and `default`?